main.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import glob
+import time
+
+from archive_converter import ArchiveMessageIndex
+from archive_updater import fill_message_directory
+from gerrit import upload_all_comments, get_gerrit_rest_api, Gerrit, find_and_label_all_revision_ids
+from git import GerritGit
+from message_dao import MessageDao
+from patch_parser import map_comments_to_gerrit, parse_comments, Patch, Patchset
+from setup_gmail import Message
+from typing import List, Set, Tuple
+
+EPOCH_HASH = 'cc49e216e3fdff0ffed7675dc7215aba5e3d05cc'
+GIT_PATH = '../linux-kselftest/git/0.git'
+FILE_DIR = 'index_files'
+
+def remove_files(file_dir : str):
+    files = glob.glob(f'{file_dir}/*')
+    for f in files:
+        os.remove(f)
+
+# returns parents, then replies as lists
+def split_parent_and_reply_messages(messages : List[Message]) -> Tuple[List[Message], List[Message]]:
+    parents : List[Message] = []
+    replies : List[Message] = []
+    for message in messages:
+            if not message.in_reply_to:
+                parents.append(message)
+                continue
+            replies.append(message)
+    return (parents, replies)
+    
+
+def main():
+    gerrit_url = 'https://linux-review.googlesource.com'
+    gob_url = 'http://linux.googlesource.com'
+    rest = get_gerrit_rest_api('gerritcookies', gerrit_url)
+    gerrit = Gerrit(rest)
+    gerrit_git = GerritGit(git_dir='gerrit_git_dir',
+                           cookie_jar_path='gerritcookies',
+                           url=gob_url, project='linux/kernel/git/torvalds/linux', branch='master')
+    message_dao = MessageDao()
+    archive_index = ArchiveMessageIndex(message_dao)
+    last_hash : str
+    # Check if database is populated
+    if(archive_index.size() == 0):
```

Comment:
Perhaps I'm overlooking something, but I think we can avoid this big `if` that has a good amount of overlap with the main loop.

Could we refactor this code so the fill+update+clear steps only appear in the main loop?